testing/deployer/topology/compile.go

```diff
@@ -13,32 +13,42 @@ import (
 	"sort"
 
 	"github.com/google/go-cmp/cmp"
+	"golang.org/x/exp/maps"
+
+	"github.com/hashicorp/go-hclog"
+
 	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v2beta1"
 	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
 	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v2beta1"
 	"github.com/hashicorp/consul/proto-public/pbresource"
-	"github.com/hashicorp/go-hclog"
-	"golang.org/x/exp/maps"
-
 	"github.com/hashicorp/consul/testing/deployer/util"
 )
 
 const DockerPrefix = "cslc" // ConSuLCluster
 
 func Compile(logger hclog.Logger, raw *Config) (*Topology, error) {
-	return compile(logger, raw, nil)
+	return compile(logger, raw, nil, "")
 }
 
 func Recompile(logger hclog.Logger, raw *Config, prev *Topology) (*Topology, error) {
```

Comment:
Why is testingID string not added to the parameters required for Recompile function?

Reply:
I would not want to expose the `testingID` parameter in an exported function yet, as it is just there for the package tests to use.

Possibly if deployer-meta-tests get written for stuff outside of the `topology` package we might revisit this. It seemed easiest to keep this testing detail internal to this package to start.